test/e2e/apps/statefulset.go

```diff
@@ -1579,7 +1584,8 @@ var _ = SIGDescribe("StatefulSet", func() {
 
```

Comment:
I can see we are doing `waitForStatus` in the other tests, but isn't there a chance for a flake as well? Shouldn't we try to confirm we have the correct set of replicas before we check they are ready?

Reply:
Good catch, yes we should definitely do a similar wait as in the other, lemme fix it...